### PR TITLE
fix #14667 and #15117

### LINKED ISSUE #14667: Visual Studio(VCC) 2019 cannot compile empty tuple generated by Nim
It works with gcc.

### Command line

```bash
nim c -r --cc:vcc bugtup
```

### Example
```nim
var bugTup = ()

proc main() =
  debugEcho bugTup
  
main()
```

### Current Output
```
C:\Users\AndriLim\nimcache\bugtup_d\@mbugtup.nim.c(64): error C2059: syntax error: '}'
Error: execution of an external compiler program 'cl.exe /c /nologo   /IF:\projects\status-nim\lib /IF:\projects\bug /FoC:\Users\AndriLim\nimcache\bugtup_d\@mbugtup.nim.c.obj C:\Users\AndriLim\nimcache\bugtup_d\@mbugtup.nim.c' failed with exit code: 2
```

### Expected Output
```
()
```

The compiler produce this C code and VCC 2019 seems doesn't like the `{}`.

```C
N_LIB_PRIVATE tyTuple__qERWGavQjzug69cyzEYP39bQ bugTup__XaG7QrFzC9aZhBgUBoCX3SQ = {};
```

```
$ nim -v
Nim Compiler Version 1.3.5 [Windows: amd64]
Compiled at 2020-06-10
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 8bbdb8f43fb0da04d4b75bca011662f690972bfe
active boot switches: -d:release
```

### LINKED ISSUE #15117: Incorrect codegen for VCC compiler
Nim compiler's codegen produces incorrect code for VCC.

### Example
```nim
let a: array[0, byte] = []

proc someProc(a: openarray[byte]): int =
  len(a)

when isMainModule:
  echo someProc(a)
```

### Current Output
```
nim c --cc:vcc emptyarray.nim

Hint: used config file 'C:\Projects\nim-beacon-chain\vendor\nimbus-build-system\vendor\Nim\config\nim.cfg' [Conf]
Hint: used config file 'C:\Projects\nim-beacon-chain\vendor\nimbus-build-system\vendor\Nim\config\config.nims' [Conf]
Hint: system [Processing]
Hint: widestrs [Processing]
Hint: io [Processing]
Hint: emptyarray [Processing]
CC: emptyarray.nim
@memptyarray.nim.c
nimcache\emptyarray_d\@memptyarray.nim.c(62): error C2059: syntax error: '}'
```

### Expected Output
```
0
```

### Additional Information
MinGW compiler works fine with produced c code.

```
Nim Compiler Version 1.2.2 [Windows: amd64]
Compiled at 2020-06-18
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: be34b5abe23e027b71a3a9da67037aac7afea6bc
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by bung87**
fix #14667 
fix #15117

### CODE REVIEW

File: compiler/ccgexprs.nim [hunk 1]
```diff
@@ -3200,12 +3200,16 @@ proc getDefaultValue(p: BProc; typ: PType; info: TLineInfo; result: var Rope) =
...
+    if p.vccAndC and t.len == 0:
+      result.add "0"
     for i in 0..<t.len:
       if i > 0: result.add ", "
       getDefaultValue(p, t[i], info, result)
     result.add "}"
   of tyArray:
     result.add "{"
+    if p.vccAndC and t.len == 0:
+      result.add "0"
```
Comment (Araq) [inline compiler/ccgexprs.nim:3212]: What if it's for `array[0, SomeObject]`? 0 is not a valid element then.

File: compiler/ccgexprs.nim [hunk 2]
```diff
@@ -3329,6 +3333,8 @@ proc genConstObjConstr(p: BProc; n: PNode; isConst: bool; result: var Rope) =
 
 proc genConstSimpleList(p: BProc, n: PNode; isConst: bool; result: var Rope) =
   result.add "{"
+  if p.vccAndC and n.len == 0:
+    result.add "0"
```
Comment (Araq) [inline compiler/ccgexprs.nim:3337]: How do you know here that `0` is a valid element for all the constructs that happen to use `genConstSimpleList`?

Comment (bung87) [reply compiler/ccgexprs.nim:3337]: how to simulate empty varargs?  seems it never be.

**Comment by bung87** [comment]
well, now `gcc` runs fails

**Pull request closed without merging**